portal/src/components/kinetic-form/widgets/signature.js

```diff
@@ -130,41 +194,115 @@ const SignatureComponent = forwardRef(
+                      .then(url => setImageUrl(url))
+                  }
+                >
+                  <div className="mb-2 font-semibold text-gray-900">
+                    <SignaturePad.Label>{signaturePadLabel}</SignaturePad.Label>
+                  </div>
+                  <SignaturePad.Control
+                    className={clsx(
+                      'border border-primary-400 bg-gray-100 relative rounded-2.5xl transition-all',
+                      'hover:bg-primary-100',
+                      'focus-within:ring focus-within:ring-secondary-400',
+                    )}
+                  >
+                    <SignaturePad.Segment />
+                    <SignaturePad.Guide
+                      className={clsx(
+                        'h-[12.5rem] relative rounded-2.5xl',
+                        'border-primary-400',
+                      )}
+                    >
+                      <SignaturePad.ClearTrigger
+                        className="absolute top-2 right-2 mr-3 mt-3"
+                        onClick={reset}
+                      >
+                        <Icon name="refresh" aria-label="reset"></Icon>
+                      </SignaturePad.ClearTrigger>
+                      <div className="flex justify-between items-center absolute bottom-4 left-5 right-6 border-t-2 border-gray-400"></div>
+                    </SignaturePad.Guide>
+                  </SignaturePad.Control>
+                </SignaturePad.Root>
+              </Tabs.Content>
+              <Tabs.Content value="type">
+                <div className="mb-4">
+                  <label
+                    htmlFor="fullName"
+                    className="block text-sm font-semibold text-gray-900 mb-2"
                   >
-                    <Icon name="refresh" aria-label="reset"></Icon>
-                  </SignaturePad.ClearTrigger>
-                  <div className="flex justify-between items-center absolute bottom-4 left-5 right-6 border-t-2 border-gray-400"></div>
-                </SignaturePad.Guide>
-              </SignaturePad.Control>
-            </SignaturePad.Root>
+                    {fullNameLabel}
+                  </label>
+                  <input
+                    type="text"
+                    value={fullName}
+                    onChange={e => setFullName(e.target.value)}
+                    className="font-semibold"
+                  />
+                </div>
+                <hr className="mb-4 mt-6 border-primary-400" />
+                <div className="grid grid-cols-2 gap-4 radio-btn">
+                  {signatureFonts.map(style => (
+                    <label
+                      key={style.font}
+                      className={`flex items-center justify-between border rounded-2.5xl p-4 w-[18rem] h-16 cursor-pointer transition-all`}
+                      onClick={() => setSelectedStyle(style.font)}
+                    >
+                      <span
```

Comment:
When the typed in name is long, the font previews wrap to multiple lines, which doesn't happen with the actual signature that gets saved. And if the name wraps to 3+ lines, then it overflows the container. You'll need to chat with Vivianne for how she wants this to work in these cases. We can probably just use ellipses to limit the content since it's just a preview.